framework/backoff_factory.go

```diff
@@ -0,0 +1,13 @@
+package framework
+
+import (
+	"github.com/cenk/backoff"
+)
+
+func DefaultBackOffFactory() func() backoff.BackOff {
+	return func() backoff.BackOff {
+		b := backoff.NewExponentialBackOff()
+		b.MaxElapsedTime = 0
+		return backoff.WithMaxTries(b, 7)
```

Comment:
7 is just a magic number I made up now. We want to have a default backoff for the framework operations that delays exponentially and stops retrying at some point. Operators can just use this and we can adapt the values in one place instead of 15.